transfer_test.go

```diff
@@ -342,3 +350,164 @@ func generateSnapshotFiles(t *testing.T) (repoSnapshotDir string) {
 	generateTestRepoSnapshotFile(t, tests.RtRepo1, repoSnapshotFile)
 	return
 }
+
+func TestTransferConfigMerge(t *testing.T) {
+	cleanUp := initTransferTest(t)
+	defer cleanUp()
+
+	targetServicesManager, err := rtutils.CreateServiceManager(targetServerDetails, -1, 0, false)
+	assert.NoError(t, err)
+
+	// Create project on Source server
+	deleteProject := createTestProject(t)
+	if deleteProject != nil {
+		defer func() {
+			assert.NoError(t, deleteProject())
+		}()
+	}
+
+	// Execute transfer-config-merge
+	assert.NoError(t, artifactoryCli.WithoutCredentials().Exec("transfer-config-merge",
+		inttestutils.SourceServerId, inttestutils.TargetServerId, "--include-repos="+tests.DockerRemoteRepo, "--include-projects="+tests.ProjectKey))
+
+	// Validate that repository transferred to target:
+	targetAuth, err := targetServerDetails.CreateArtAuthConfig()
+	assert.NoError(t, err)
+	if assert.NoError(t, rtutils.ValidateRepoExists(tests.DockerRemoteRepo, targetAuth)) {
+		defer func() {
+			assert.NoError(t, targetServicesManager.DeleteRepository(tests.DockerRemoteRepo))
+		}()
+	}
+	// Validate that project transferred to target:
+	targetAccessManager, err := rtutils.CreateAccessServiceManager(serverDetails, false)
+	assert.NoError(t, err)
+	projectDetails, err := targetAccessManager.GetProject(tests.ProjectKey)
+	if assert.NoError(t, err) && assert.NotNil(t, projectDetails) {
+		defer func() {
+			assert.NoError(t, targetAccessManager.DeleteProject(tests.ProjectKey))
+		}()
+	}
+
+	// Validate no conflicts between source and target repositories
+	configMergeCmd := transferconfig.NewTransferConfigMergeCommand(serverDetails, targetServerDetails).
```

Comment:
It is best to construct the command using the CLI interface (with arguments and flags). This is beucase in the past we have seen bugs that are related to the CLI command construction, which the tests didn't catch.